### PR TITLE
Permissions notifications

### LINKED ISSUE #64: Sort out permissions
# Goal

Ensure only the necessary permissions are requested and have a popup when the user opens the app the first time asking for these

# Plan

1. Investigate permissioning on Android and iOS

### PR DESCRIPTION
**Pull request by C-Begley**
# Details

- Asks user for permissions
- If user denies permission, alert explaining permission is needed, with link to settings or denied
- If the user opens the app twice, denying permissions both times, Android considers permission blocked
- If permission is blocked,only alert appears and no longer asks for permissions 

# Screenshots/GIF (if appropriate)
It can sometimes be useful to add a screenshot or animated GIF to demonstrate new functionality

# Checklist

## Code
- [ ] I have added some tests
- [ ] I have run the tests locally to ensure they all pass

## Documentation
- [ ] I have updated documentation appropriately (GitHub Pages)
- [x] I have tested the updated GitHub Pages locally to ensure the changes to documentation render correctly (see this [Guidance](https://ese-peasy.github.io/PosturePerfection/guidance.html) page)

## Accessibility
- [x] Tested Accessibility on **Android**
  - [x] Any added fields can be read aloud by the screen reader
- [ ] Tested Accessibility on **iOS**
  - [ ] Any added fields can be read aloud by the screen reader

Closes #64

### CODE REVIEW

File: app/App.tsx [hunk 1]
```diff
@@ -8,8 +8,82 @@ import useCachedResources from './hooks/useCachedResources';
...
+    console.log('ios');
+    perms = [
+      PERMISSIONS.IOS.LOCATION_ALWAYS,
+      PERMISSIONS.IOS.BLUETOOTH_PERIPHERAL
+    ];
+  }
+  checkAllPermissions(perms).then((reqs) => {
+    if (reqs.length > 0) {
+      console.log('Requesting Permissions');
+      console.log(reqs);
```
Comment (Ashwin-MJ) [inline app/App.tsx]: Maybe get rid of these and the other logs?

File: app/App.tsx [hunk 2]
```diff
@@ -8,8 +8,82 @@ import useCachedResources from './hooks/useCachedResources';
...
+                }
+              ]
+            );
+            break;
+          }
+        }
+      });
+    }
+  });
+};
```
Comment (Ashwin-MJ) [inline app/App.tsx]: All of this could probably be put in a separate file within `src/` so we only need to import `checkAndGetAllPermissions`?

**Review by miklasr — DISMISSED** [review]
I've checked different buttons for accepting/declining location permissions. Everything seems to work as expected.

Nice stuff!

**Review by Ashwin-MJ — CHANGES_REQUESTED** [review]
If I have denied permissions and then have to enable it via settings, when i re-open the app it doesn't seem to actually start scanning for beacons (it doesn't detect anything)

**Comment by C-Begley** [comment]
> If I have denied permissions and then have to enable it via settings, when i re-open the app it doesn't seem to actually start scanning for beacons (it doesn't detect anything)

I think if you don't close the app between settings it won't work. I'll try and change that

**Pull request closed without merging**